flink-python/pyflink/table/udf.py

```diff
@@ -149,13 +149,12 @@ def __init__(self, func, input_types, result_type, deterministic=None, name=None
         self._deterministic = deterministic if deterministic is not None else (
             func.is_deterministic() if isinstance(func, UserDefinedFunction) else True)
 
-    @property
-    def _judf(self):
+    def _judf(self, t_env):
```

Comment:
We can replace the "t_env" parameter with the boolean value "is_blink_planner" here. The whole table_environment is unnecessary. After that we can remove the public property "is_blink_planner" and use "_is_blink_planner" instead.